src/vs/workbench/contrib/tasks/browser/runAutomaticTasks.ts

```diff
@@ -106,22 +105,22 @@ export class RunAutomaticTasks extends Disposable implements IWorkbenchContribut
 					});
 				}
 				if (resultElement.configurations) {
-					forEach(resultElement.configurations.byIdentifier, (configedTask) => {
-						if (configedTask.value.runOptions.runOn === RunOnOptions.folderOpen) {
+					for (const configuredTask of Object.entries(resultElement.configurations.byIdentifier)) {
```

Comment:
This should use `resultElement.configurations.byIdentifier.values()` so you don't need to use `[1]`